### ISSUE TITLE
Dynamic plot adding broken?

### ISSUE THREAD

**Issue by bkueng** [bokeh/bokeh#6407]
Looks like on current master (fb5c103e3a63935), dynamic plot adding is broken.

Using the following script:
```
from bokeh.plotting import figure, show, output_file
from bokeh.client import push_session
from bokeh.io import curstate, curdoc
from bokeh.models.layouts import VBox
curstate().autoadd = False

import time
import numpy as np

x = np.linspace(0, 4*np.pi, 100)
y = np.sin(x)

# create first plot
plot = figure( y_range=(-1.2, 1.2) )
plot.line(x,y,line_width=2, legend="x")

# add plot to the layout
layout = VBox(children=[plot])

curdoc().add_root(layout)
plot_counter = 0

def callback():
    global plot_counter
    if plot_counter > 1:
        return
    plot_counter += 1
    # create additional plots
    plot2 = figure( y_range=(-1.2, 1.2) )
    plot2.line(x,y,line_width=2, legend="x")

    # add additional plots to the layout
    layout.children.append(plot2)
    print("adding plot")

curdoc().add_periodic_callback(callback, 5000)
```
the first additional plot gets added after 5s, but with missing toolbar, while the second plot after 5 more seconds does not get added/displayed at all. When resizing the browser window, all 3 plots show up, but the toolbar of the first plot gets moved to the left side of the plot.

It works as expected with 0.12.5

Tested on Firefox 53

**Comment by bryevdv** [comment]
@bkueng Thanks for the report. Your code seems to work fine with #6384 which fixes a raft of layout-related problems. We are working hard to get it in to `0.12.6` I also believe this is covered by one one of the issues linked there, so I am closing this as a duplicate. If you have a chance to help test with that PR and also with RCs once it is merged it would be very helpful.

**Comment by bkueng** [comment]
Ah sorry, I did not find anything related. Sure I'll test it.

**Comment by mattpap** [comment]
I can confirm that this is fixed by PR #6384.
